qa/workunits/windows/test_rbd_wnbd.py

```diff
@@ -360,10 +360,11 @@ def init_fs(self):
         # Retrieve the drive letter.
         cmd = (
             "powershell.exe", "-command",
-            f"(Get-Partition -DiskNumber {self.disk_number})[0].DriveLetter")
+            f"(Get-Partition -DiskNumber {self.disk_number}"
+            " | ? { $_.DriveLetter }).DriveLetter")
         result = execute(*cmd)
 
-        self.drive_letter = result.stdout.decode().strip()
+        self.drive_letter = result.stdout.decode().strip().replace('\x00', '')
```

Comment:
> While at it, we're removing null characters from the retrieved drive letter.

Where are the (multiple?) NUL characters coming from?  Isn't it just a single NUL character at the end meaning that we could just do `[0]` instead of calling `replace` here?

Reply:
It's putting a null character if no drive letter is available, here are some examples:

```
>>> execute("powershell.exe", "-command", "get-partition -disknumber 0")
CompletedProcess(args=('powershell.exe', '-command', 'get-partition -disknumber 0'), returncode=0, stdout=b'\r\n\r\n   DiskPath: \r\n\\\\?\\scsi#disk&ven_qemu&prod_harddisk#4&35424867&0&000000#{53f56307-b6bf-11d0-94f2-00a0c91efb8b}\r\n\r\nPartitionNumber  DriveLetter Offset                               Size Type                          \r\n---------------  ----------- ------                               ---- ----                          \r\n1                \x00           1048576                            100 MB IFS                           \r\n2                C           105906176                       119.39 GB IFS                           \r\n3                \x00           128297467904                       524 MB Unknown                       \r\n\r\n\r\n', stderr=b'')


>>> execute("powershell.exe", "-command", "(get-partition -disknumber 0)[0].DriveLetter")
CompletedProcess(args=('powershell.exe', '-command', '(get-partition -disknumber 0)[0].DriveLetter'), returncode=0, stdout=b'\x00\r\n', stderr=b'')
```

This is the closest documentation that I could find: https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_special_characters?view=powershell-7.3#null-0

Reply:
I see, unfortunately this doesn't make it any less confusing.  How about amending the `len(self.drive_letter) != 1` test to something like `if not self.drive_letter.isalpha() or len(self.drive_letter) != 1`?  Otherwise, let's add a comment highlighting the possible `\x00\r\n` output.

Reply:
Sure, I'll add an inline comment. The ``if not self.drive_letter.isalpha()`` extra check can be useful too.

Reply:
FWIW what I had in mind was `if not self.drive_letter.isalpha()` instead of `replace('\x00', '')`.  `\0` is definitely not alphanumeric so one of them is redundant and it's better to keep one that does a test instead of a modification.